actionpack/lib/action_controller/metal/content_security_policy.rb

```diff
@@ -28,9 +28,19 @@ def content_security_policy(enabled = true, **options, &block)
         end
       end
 
-      def content_security_policy_report_only(report_only = true, **options)
+      def content_security_policy_report_only(enabled = true, **options, &block)
         before_action(options) do
-          request.content_security_policy_report_only = report_only
+          if block_given?
+            policy = current_content_security_policy
+            yield policy
```

Comment:
Won't this mutate the policy for the CSP header if both are being generated?

Reply:
You are right. I've reproduced with a test.

Reply:
@pixeltrix I've addressed this https://github.com/rails/rails/pull/40525/commits/c6d9e0005e99d9506863d9f9f180fd394c481be6